### ISSUE TITLE
Feature request: show install time for each package

### ISSUE THREAD

**Issue by Afoucaul** [astral-sh/uv#7074]
When you run `uv sync` or `uv pip install -r requirements.txt`, a bunch of packages get installed.
You get a spinner, then a complete list of the packages installed; example:

```bash
$ uv sync
Using Python 3.8.12
Creating virtualenv at: .venv
Resolved 53 packages in 23ms
Prepared 12 packages in 7ms
Installed 14 packages in 53ms
 + anyio==4.4.0
 + certifi==2024.8.30
 + charset-normalizer==3.3.2
 + exceptiongroup==1.2.2
 + h11==0.14.0
 + httpcore==1.0.5
 + httpx==0.27.2
 + idna==3.8
 + requests==2.32.3
 + sniffio==1.3.1
 + typing-extensions==4.5.0
 + urllib3==2.2.2
```

While the total install time is a good metric to know, I would like to know the install time of each specific package. Even though this doesn't directly correlate with wall time, it would be great to identify the slowest packages to install and thus reducing install and build times.

A concrete example: pandas 1.3.5 takes forever to install. But if my project uses `pandas==1.3.5` and anything else, say `rich`, then my `uv sync --no-cache` output looks like follows - how can I know it's pandas that made the sync take so long?

```bash
$ uv sync --no-cache
Using Python 3.8.12
Creating virtualenv at: .venv
Resolved 11 packages in 1ms
   Built pandas==1.3.5
Prepared 10 packages in 3m 28s
Installed 10 packages in 131ms
 + markdown-it-py==3.0.0
 + mdurl==0.1.2
 + numpy==1.24.4
 + pandas==1.3.5
 + pygments==2.18.0
 + python-dateutil==2.9.0.post0
 + pytz==2024.1
 + rich==13.8.0
 + six==1.16.0
 + typing-extensions==4.12.2
```

I can identify that it's pandas that takes the longest to install, because it's where the spinner hangs for the longest, but I can't get that retrospectively in the logs.
Once I've identified that, I was able to upgrade it to a more recent version, that provides pre-built wheels which makes the install much faster (from 3min30 to 2sec) - so identifying which specific package takes long to install has a huge value.

I'd be happy with some `--annotate-with-time` flag if that works for you!

**Comment by zanieb** [comment]
Makes sense, though I'm not sure how straight-forward it would be to add.

**Comment by Afoucaul** [comment]
`--timings` sounds perfect!

**Comment by jaymegordo** [comment]
+1 on this, would be extremely helpful to have especially in CICD pipelines. Eg you might have some packages slowing down your installs (on slower machines eg github actions) and kinda forget about them, not sure if its worth looking into removing them etc.

**Comment by farzbood** [comment]
any update on that --timings ?

**Comment by farzbood** [comment]
> any update on that --timings ?

A real life situation that needed to be analyzed regarding syncing latency e.g `zensical `too long to download and install as following:
PS D:\pyprojects\Packages\fastapi-nlweb-widget-manuals\fastapi-nlweb-widget-docs> uv sync
Resolved 12 packages in 1.29s
░░░░░░░░░░░░░░░░░░░░ [0/11] Installing wheels...                                          warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
Installed 11 packages in 5m 21s
 + click==8.4.2
 + colorama==0.4.6
 + deepmerge==3.0
 + jinja2==3.1.6
 + markdown==3.10.3
 + markupsafe==3.0.3
 + pygments==2.21.0
 + pymdown-extensions==11.0.2
 + pyyaml==6.0.3
 + tomli==2.4.1
 + zensical==0.0.57
but there's no way to find what package caused the abnormal latency!
I've opened an [issue in zensical repo](https://github.com/zensical/zensical/issues/881) first, contains complete details and logs.
**P.S**
I've projects on the same platform with numerous dependencies (10-20 X more than this project) with no latency.
